custom_components/zaptec/__init__.py

```diff
@@ -323,14 +325,59 @@ async def _async_update_data(self) -> None:
 
                 # Fetch updates
                 _LOGGER.debug("Polling from Zaptec")
-                await self.zaptec.update_states()
+                await self.zaptec.poll(
+                    self.updateable_objects, state=True, info=True, firmware=True
+                )
 
         except ZaptecApiError as err:
             _LOGGER.exception(
                 "Fetching data failed: %s: %s", type(err).__qualname__, err
             )
             raise UpdateFailed(err) from err
 
+    async def _trigger_poll(self, obj: ZaptecBase) -> None:
+        """Trigger a poll update sequence for the given object or all objects.
+
+        This sequence is useful to ensure that the state is fully synced after a
+        HA initiated update.
+        """
+
+        what = {obj.id}
+        if isinstance(obj, Installation):
+            delays = ZAPTEC_POLL_INSTALLATION_TRIGGER_DELAYS
+            kw = {"state": True, "info": True}
+        else:
+            delays = ZAPTEC_POLL_CHARGER_TRIGGER_DELAYS
+            kw = {"state": True}
+
+        # Calculcate the deltas for the delays. E.g. [2, 5, 10] -> [2, 3, 5]
+        deltas = [b - a for a, b in zip([0] + delays[:-1], delays)]
+
+        for i, delay in enumerate(deltas, start=1):
```

Comment:
At first glance, this looked like a bug since we have a `delays` list that shouldn't be used directly for the sleep, while we are using `delay` here. (deltas and delays being the same length with 4/6 letters in common certainly doesn't help)

While I realize it is correct as written, wouldn't it be better to use `delta` as the variable in the loop?

Reply:
The onliner definition of deltas is fine by me, I'm asking if
```suggestion
        for i, delta in enumerate(deltas, start=1):
```
(and using delta in the rest of the for loop as well) is easier to read? For me at least, `for delta in deltas` feels more natural, especially when there is already a `delays` list within the same code block.

Reply:
yes, I agree.

You have a keen eye for details like this that evidently pass my eyes. Thanks. 👍

Reply:
Fixed